src/Compilers/CSharp/Test/Semantic/Semantics/InterpolationTests.cs

```diff
@@ -17992,5 +18012,277 @@ public void InterpolatedStringInForeach_HasErrors()
+                Diagnostic(ErrorCode.ERR_ExpectedInterpolatedString, @"""""").WithLocation(5, 14),
+                // (6,18): error CS1510: A ref or out value must be an assignable variable
+                //         ByIn(ref "");
+                Diagnostic(ErrorCode.ERR_RefLvalueExpected, @"""""").WithLocation(6, 18),
+                // (7,17): error CS8156: An expression cannot be used in this context because it may not be passed or returned by reference
+                //         ByIn(in "");
+                Diagnostic(ErrorCode.ERR_RefReturnLvalueExpected, @"""""").WithLocation(7, 17),
+                // (8,18): error CS1510: A ref or out value must be an assignable variable
+                //         ByIn(out "");
+                Diagnostic(ErrorCode.ERR_RefLvalueExpected, @"""""").WithLocation(8, 18)
+            );
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/66235")]
+        public void ConvertStringToInterpolatedStringHandlerError_DefaultInterpolatedStringHandler_ByOut()
+        {
+            var code = """
+                class C
+                {
+                    void M()
+                    {
+                        ByOut("");
+                        ByOut(ref "");
+                        ByOut(in "");
+                        ByOut(out "");
+                    }
+
+                    void ByOut(out System.Runtime.CompilerServices.DefaultInterpolatedStringHandler s) { s = default; }
+                }
+                """;
+
+            CreateCompilation(code, targetFramework: TargetFramework.Net60).VerifyDiagnostics(
+                // (5,15): error CS1620: Argument 1 must be passed with the 'out' keyword
+                //         ByOut("");
+                Diagnostic(ErrorCode.ERR_BadArgRef, @"""""").WithArguments("1", "out").WithLocation(5, 15),
+                // (6,19): error CS1510: A ref or out value must be an assignable variable
+                //         ByOut(ref "");
+                Diagnostic(ErrorCode.ERR_RefLvalueExpected, @"""""").WithLocation(6, 19),
+                // (7,18): error CS8156: An expression cannot be used in this context because it may not be passed or returned by reference
+                //         ByOut(in "");
+                Diagnostic(ErrorCode.ERR_RefReturnLvalueExpected, @"""""").WithLocation(7, 18),
+                // (8,19): error CS1510: A ref or out value must be an assignable variable
+                //         ByOut(out "");
+                Diagnostic(ErrorCode.ERR_RefLvalueExpected, @"""""").WithLocation(8, 19)
+            );
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/66235")]
+        public void ConvertStringToInterpolatedStringHandlerError_CustomInterpolatedStringHandler_ByOut()
+        {
+            var code = """
+                class C
+                {
+                    void M()
+                    {
+                        ByOut("");
```

Comment:
Please add examples in these tests with non-string arguments, like local references or ints. In general, I'm not convinced about the condition being as broad as you have it in this PR: I could see erroring specifically when passed a string literal, but not when passed something else, like a local variable.

Reply:
Added a test with local variables instead of literals. I think, for the purpose of this PR it is enough, especially considering that there are 2 other non-string-literal tests in the different part of the codebase which I had to modify earlier, but now are back to their original state